### ISSUE TITLE
systemd-repart fails when no initrd stage and systemd is PID1

### ISSUE THREAD

**Issue by tewarid** [systemd/systemd#24465]
### Component

systemd-repart

### Is your feature request related to a problem? Please describe

In a systemd with not initrd and systemd as PID1, systemd-repart fails

```log
sh-5.1# systemctl -l status systemd-repart          
x systemd-repart.service - Repartition Root Disk
     Loaded: loaded (/lib/systemd/system/systemd-repart.service; static)
     Active: failed (Result: exit-code) since Mon 2022-08-08 09:10:05 UTC; 1min 27s ago
       Docs: man:systemd-repart.service(8)
    Process: 100 ExecStart=/bin/systemd-repart --dry-run=no (code=exited, status=1/FAILURE)
   Main PID: 100 (code=exited, status=1/FAILURE)

Aug 08 09:10:05 raspberrypi0-wifi systemd-repart[100]: Failed to determine backing device of /: No such file or directory
```

### Describe the solution you'd like

systemd-repart should work when no initrd stage and systemd is PID1.

### The systemd version you checked that didn't have the feature you are asking for

251

**Comment by tewarid** [comment]
Forgot to mentions that `/etc/repart.d/linux-generic.conf` contains

```conf
[Partition]
Type=linux-generic
```

Also, `systemd-repart --dry-run=no` is able to grow root partition from a root shell

```log
sh-5.1# systemd-repart --dry-run=no 
Configuration file /etc/repart.d/linux-generic.conf is marked world-writable. Please remove world writability permission bits. Proceeding anyway.
TYPE       LABEL      UUID        FILE        NODE               SIZE    PADDING
linux-g... Linux f... 2da56c1c... linux-ge... /dev/mmc... 317.1M -... 7.0G -> 0B
ebd0a0a... Microso... cd7df36b... -           /dev/mmc...       49.9M       2.0M
                                                             S = 7.4G   S = 2.0M

 XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX-
 |- Microsoft basic data                                                       
 `- linux-generic.conf                                                         

Applying changes.
Successfully discarded gap at beginning of disk.
Growing existing partition 1.
Writing new partition table.
Telling kernel to reread partition table.
All done.
```

**Comment by yuwata** [comment]
Fix is waiting in #24475.
